.github/workflows/test.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow will be triggered by a GitHub pull-request and pushes to main.
+# It will verify that the code compiles, that the tests pass and that
+# clippy does not complain (too much) and finally that the code is formatted
+# according to cargo fmt. If it was a push to main, then it will also trigger a
+# build, and a deploy to dev.
+---
+name: Test
+
+on:
+  pull_request:
+    branches: ["*"]
+  push:
+    branches: ["main"]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: stable
+          components: clippy, rustfmt
+          cache: true
+
+      - name: Rustfmt Check
+        uses: actions-rust-lang/rustfmt@v1
+
+      # - name: Lint code
```

Comment:
why is clippy commented out

Reply:
I copied this from another workflow and that clippy action is abandoned, so I didn't want to include it here. I didn't search for any replacement steps. We could just run it directly, ie `cargo clippy` :thinking:

Reply:
even if the action is abandoned it still works and the advantage of that action is inline annotation as opposed to none when just running `cargo clippy`

Reply:
Unfortunately it will result in some warning regarding deprecated functionality in GitHub, so I would rather just have it working even if that means no annotations.